test/parallel/test-readable-single-end.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const common = require('../common');
+const { Readable } = require('stream');
+
+// This test ensures that there will not be an additional empty 'readable'
+// event when stream has ended (only 1 event signalling about end)
+
+const r = new Readable({
+  read: () => {},
+});
+
+r.push(null);
+
+r.once('readable', common.mustCall());
```

Comment:
Should this not also just be `on`?

Reply:
Thanks, idk why did I think it was okay to put once on both of them. :thinking: